### DISCUSSION TITLE
What is the best practice to output CSS for a library?

### DISCUSSION THREAD

**Discussion by Francoshum95** [tailwindlabs/tailwindcss#10921] [Help] [answered]
How can I prevent conflicts with class names and global scope styles used by the application? 
Any suggestions or advice

**Comment by wongjn** [comment]
You could try using the [`important`](https://tailwindcss.com/docs/configuration#selector-strategy) configuration option. Otherwise, if you are running Tailwind CSS as a PostCSS plugin, you could add another plugin after that scopes the rules as needed.

**Reply by wongjn** [reply]
> Thanks, @wongjn ! Are there any PostCSS plugins you would recommend? Do you recommend disabling preflight and setting a prefix to adjust the class name ?[^1]

No, I don't have any PostCSS plugins I'd recommend – I normally roll my own in this situation. I don't have enough context to be able to recommend nor not recommend disabling preflight for your situation – this is something you will have to review.

[^1]: https://github.com/tailwindlabs/tailwindcss/discussions/10921#discussioncomment-5489274

**Comment by Francoshum95** [comment]
Thanks, @wongjn ! Are there any PostCSS plugins you would recommend? 
Do you recommend disabling preflight and setting a prefix to adjust the class name ?

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/10921#discussioncomment-5491050 to keep the reply chain together.

**Comment by olets** [comment, marked as answer]
> The prefix option allows you to add a custom prefix to all of Tailwind’s generated utility classes. This can be really useful when layering Tailwind on top of existing CSS where there might be naming conflicts.

https://tailwindcss.com/docs/configuration#prefix

**Reply by Francoshum95** [reply]
Thanks @olets , I think this is the most suitable choice. Although I was using CSS Modules to create a class name with a hashed suffix, I discovered that it can't be utilized as a dynamic class name.
